cpp/src/replace/replace.cu

```diff
@@ -182,51 +192,107 @@ void replace_nulls_with_column(gdf_size_type size, Type* out_data, gdf_valid_typ
-                                            (d_col_valid),
-                                            static_cast<const col_type*>(d_new_value)
-                                            );
-    } else if(new_values_length == nrows) {
-      replace_nulls_with_column<<<grid_size, BLOCK_SIZE>>>(nrows,
-                                            static_cast<col_type*>(d_col_data),
-                                            (d_col_valid),
-                                            static_cast<const col_type*>(d_new_value)
-                                            );
-      
-    }
+
+    replace_nulls_with_column<<<grid_size, BLOCK_SIZE>>>(nrows,
+                                          static_cast<const col_type*>(d_in_data),
+                                          (d_in_valid),
+                                          static_cast<const col_type*>(d_replacement),
+                                          static_cast<col_type*>(d_out_data)
+                                          );
+
+  }
+};
+
+
+/* --------------------------------------------------------------------------*/
+/** 
+ * @brief Functor called by the `type_dispatcher` in order to invoke and instantiate
+ *        `replace_nulls` with the apropiate data types.
+ */
+/* ----------------------------------------------------------------------------*/
+struct replace_nulls_scalar_kernel_forwarder {
+  template <typename col_type>
+  void operator()(gdf_size_type    nrows,
+                  void*            d_in_data,
+                  gdf_valid_type*  d_in_valid,
+                  const void*      replacement,
+                  void*            d_out_data)
+  {
+    const size_t grid_size = nrows / BLOCK_SIZE + (nrows % BLOCK_SIZE != 0);
+
+    auto t_replacement = static_cast<const col_type*>(replacement);
+    void *d_replacement = NULL;
+    RMM_TRY(RMM_ALLOC(&d_replacement, sizeof(col_type), 0));
+    CUDA_TRY(cudaMemcpy(d_replacement, t_replacement, sizeof(col_type),
+                        cudaMemcpyHostToDevice));
+
+    replace_nulls_with_scalar<<<grid_size, BLOCK_SIZE>>>(nrows,
+                                          static_cast<const col_type*>(d_in_data),
+                                          (d_in_valid),
+                                          static_cast<const col_type*>(d_replacement),
+                                          static_cast<col_type*>(d_out_data)
+                                          );
   }
 };
 
+
+
 } //end anonymous namespace
 
-gdf_error gdf_replace_nulls(gdf_column* col_out, const gdf_column* col_in)
+
+namespace cudf {
+
+gdf_column replace_nulls(const gdf_column& input,
+                         const gdf_column& replacement)
 {
-  GDF_REQUIRE(col_out->dtype == col_in->dtype, GDF_DTYPE_MISMATCH);
-  GDF_REQUIRE(col_in->size == 1 || col_in->size == col_out->size, GDF_COLUMN_SIZE_MISMATCH);
+  CUDF_EXPECTS(input.dtype == replacement.dtype, "Data type mismatch");
```

Comment:
Need to add a check for when `input.size == 0` and just return `cudf::empty_like(input)`.

Reply:
I think this is done.